lib/symboldatabase.cpp

```diff
@@ -3238,18 +3237,17 @@ void SymbolDatabase::addNewFunction(Scope **scope, const Token **tok)
             while (tok1->next())
                 tok1 = tok1->next();
             *scope = nullptr;
-            *tok = tok1;
-            return;
+        } else {
+            (*scope)->nestedList.push_back(newScope);
+            *scope = newScope;
         }
-
-        (*scope)->nestedList.push_back(newScope);
-        *scope = newScope;
-        *tok = tok1;
+    } else if (tok1->tokAt(-2) && Token::Match(tok1->tokAt(-2), "= default|delete ;")) {
+        scopeList.pop_back();
     } else {
         scopeList.pop_back();
         *scope = nullptr;
-        *tok = nullptr;
```

Comment:
This `*tok = nullptr` was the curlpit why the code after `= delete` wan't analysed at all. Which gave a false sense of security, because cppcheck complained about nothing, but also didn't notify me that something was broken.

I removed the nullptr to prevent it from being silent on future issues, but shouldn't it throw some InvalidAst or something? Similar for a few lines up, with the comment `// syntax error?`. There as well, `tok` ends up as the last token, effectively skipping the analysis.

Also, I'm not sure about the `*scope = nullptr` in this case. Why not keep the existing scope we already had?

Reply:
By the way: two unit tests entered this `else`: `TestClass::testGetFileInfo` and the 3rd case in `TestValueFlow::valueFlowCrashIncompleteCode`.
For this second case, what's the goal of the test? Just that it doesn't crash? Because the faulty behaviour was present here, but the unit test didn't show it...

Reply:
If we run into significant problems then yes some kind of internal error should be generated.

Reply:
> Just that it doesn't crash? 

Yes, its to make sure we dont crash on incomplete code.